### DISCUSSION TITLE
Can't install / update `gh` due to expired GPG key?

### DISCUSSION THREAD

**Discussion by github-staff** [community/community#138304] [Announcements]
**TL;DR This [issue in the cli repo](https://github.com/cli/cli/issues/9569) includes a detailed update on what happened, and what you need to do about it.**

On Friday 6th September at approximately 11:17 am UTC, the GPG key used to verify our `.deb` and `.rpm` package repository contents will expire.
This will impact apt update and dnf install / update usage, which will look something like the following:
```
W: GPG error:[ https://cli.github.com/packages](https://cli.github.com/packages) stable InRelease: The following signatures were invalid: EXPKEYSIG 23F3D4EA75716059 GitHub CLI <[opensource+cli@github.com](mailto:opensource+cli@github.com)>
```

If you have not previously installed the GitHub CLI via our package repositories, there should be no impact for you.

- The CLI repo issue [Can't install / update ](https://github.com/cli/cli/issues/9569)gh[ due to expired GPG key?](https://github.com/cli/cli/issues/9569) is the best place to find updates, ask questions, and share feedback.
- Additionally, the CLI team informed users in the [2.56.0 release notes](https://github.com/cli/cli/releases/tag/v2.56.0) posted this week

## Update: Sept 4, 2026
> [!IMPORTANT]
> This discussion dates back to September 2024. If you are looking for current information about GitHub CLI Linux package signature errors after September 5, 2026, see the [September 2026 PGP key rotation announcement](https://github.com/cli/cli/issues/13118). It includes steps to update your keyring and resolve package installation or update problems.

**Comment by jinhua115** [comment]
https://github.com/orgs/community/discussions/138304#discussion-7165786

**Comment by TahmidFarzan** [comment]
Dear GitHub Support Team @github-staff,

I am writing to express my urgent need for assistance regarding my GitHub account, which has been flagged and the reason of being flag is unknown to me. This issue is severely impacting my ability to collaborate with my team, as I am unable to make pull requests and contribute effectively to our projects.

I created a support ticket (#2916054) Aug 2024 ago, but unfortunately, I have not received any response yet. I understand that the support team may be dealing with a high volume of requests, but this situation is critical for me. The inability to perform my duties is putting my job at risk, and I am growing increasingly anxious about the potential consequences.

As a committed and active GitHub user, I rely heavily on the platform for my work. I assure you that I have always adhered to the community guidelines and policies. If there is any specific reason for the flagging of my account, I would appreciate detailed information and guidance on how to resolve it promptly.

Thank you very much for your understanding and prompt assistance. I look forward to your positive response.

**Comment by hholst80** [comment]
I have fedora 39

HOW do I install this package?

```
[I] root@goblin ~/o/docker/hello-web (master) 
# dnf install gh
Copr repo for hyprland owned by solopasha                                 305  B/s | 341  B     00:01    
Errors during downloading metadata for repository 'copr:copr.fedorainfracloud.org:solopasha:hyprland':
  - Status code: 404 for https://download.copr.fedorainfracloud.org/results/solopasha/hyprland/fedora-39-x86_64/repodata/repomd.xml (IP: 3.164.230.64)
Error: Failed to download metadata for repo 'copr:copr.fedorainfracloud.org:solopasha:hyprland': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried
Ignoring repositories: copr:copr.fedorainfracloud.org:solopasha:hyprland
Last metadata expiration check: 0:58:45 ago on Tue 12 Nov 2024 05:28:53 PM CET.
Dependencies resolved.
==========================================================================================================
 Package              Architecture             Version                     Repository                Size
==========================================================================================================
Installing:
 gh                   x86_64                   2.61.0-1                    gh-cli                    13 M

Transaction Summary
==========================================================================================================
Install  1 Package

Total size: 13 M
Installed size: 48 M
Is this ok [y/N]: y
Downloading Packages:
[SKIPPED] gh_2.61.0_linux_amd64.rpm: Already downloaded                                                  
error: Verifying a signature using certificate 2C6106201985B60E6C7AC87323F3D4EA75716059 (GitHub CLI <opensource+cli@github.com>):
  1. Certificate 23F3D4EA75716059 invalid: certificate is not alive
      because: The primary key is not live
      because: Expired on 2024-09-06T11:17:19Z
  2. Key 23F3D4EA75716059 invalid: key is not alive
      because: The primary key is not live
      because: Expired on 2024-09-06T11:17:19Z
packages for the GitHub CLI                                                43 kB/s | 4.7 kB     00:00    
GPG key at https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x23F3D4EA75716059 (0x75716059) is already installed
The GPG keys listed for the "packages for the GitHub CLI" repository are already installed but they are not correct for this package.
Check that the correct key URLs are configured for this repository.. Failing package is: gh-2.61.0-1.x86_64
 GPG Keys are configured as: https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x23F3D4EA75716059
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: GPG check FAILED
[I] root@goblin ~/o/docker/hello-web (master) [1] 
# 
```


### Update

The only way I have found to install this is to download the RPM and run:


```
[I] root@goblin /tmp [1] 
# rpm -ivh --nodigest --nosignature ./gh_2.61.0_linux_amd64.rpm

Preparing...                          ################################# [100%]
Updating / installing...
   1:gh-0:2.61.0-1                    ################################# [100%]
[I] root@goblin /tmp 
# 
```

**Comment by Mohamedebraim12** [comment]
> **TL;DR This [issue in the cli repo](https://github.com/cli/cli/issues/9569) includes a detailed update on what happened, and what you need to do about it.**
> 
> On Friday 6th September at approximately 11:17 am UTC, the GPG key used to verify our `.deb` and `.rpm` package repository contents will expire.
> This will impact apt update and dnf install / update usage, which will look something like the following:
> ```
> W: GPG error:[ https://cli.github.com/packages](https://cli.github.com/packages) stable InRelease: The following signatures were invalid: EXPKEYSIG 23F3D4EA75716059 GitHub CLI <[opensource+cli@github.com](mailto:opensource+cli@github.com)>
> ```
> 
> If you have not previously installed the GitHub CLI via our package repositories, there should be no impact for you.
> 
> - The CLI repo issue [Can't install / update ](https://github.com/cli/cli/issues/9569)gh[ due to expired GPG key?](https://github.com/cli/cli/issues/9569) is the best place to find updates, ask questions, and share feedback.
> - Additionally, the CLI team informed users in the [2.56.0 release notes](https://github.com/cli/cli/releases/tag/v2.56.0) posted this week

**Reply by Mohamedebraim12** [reply]
Good to deliver

**Comment by Mohamedebraim12** [comment]
function balanceOf(address _owner) public view returns (uint256 balance)

**Comment by Mohamedebraim12** [comment]
function balanceOf(address _owner) public view returns (uint256 balance)

**Comment by Mohamedebraim12** [comment]
7d036f57fd26a53269d4b64c789cbb1f5ebccf6a

**Comment by eaaltonen** [comment]
https://github.com/cli/cli/issues/9569 mentions the key fingerprint `2C6106201985B60E6C7AC87323F3D4EA75716059`, providing (_some_) evidence that this primary key is controlled by the GitHub CLI team. On 2026-08-24, my download from https://cli.github.com/packages/githubcli-archive-keyring.gpg contained also a second primary key.

I would appreciate if the GitHub CLI team could confirm or deny in #9569 whether a second PGP primary key is used by the team. Also if second primary key is used, a separate download with a signature from the first key would be highly welcome.

**Reply by williammartin** [reply]
@eaaltonen sorry your message got lost but https://github.com/cli/cli/issues/13118 probably gives you the information you need.

**Comment by babakks** [comment]
> [!IMPORTANT]
> This discussion dates back to September 2024. If you are looking for current information about GitHub CLI Linux package signature errors after September 5, 2026, see the [September 2026 PGP key rotation announcement](https://github.com/cli/cli/issues/13118). It includes steps to update your keyring and resolve package installation or update problems.
